File: core/src/main/java/com/tek/core/converter/HashMapConverter.java
package com.tek.core.converter;

import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.extern.slf4j.Slf4j;

import javax.persistence.AttributeConverter;
import java.util.HashMap;

/**
 * Hashmap String Type Conversion.
 * <p>
 * This implementation converts a String representation of a map in a database (like plain JSON objects)
 * to a more friendly user HashMap java type.
 *
 * @author MarcoPagan
 */
@Slf4j
public class HashMapConverter implements AttributeConverter<HashMap<String, Object>, String> {

    private static final ObjectMapper objectMapper = new ObjectMapper();

    @Override
    public String convertToDatabaseColumn(HashMap<String, Object> attribute) {
        String text = null;
        try {
            text = objectMapper.writeValueAsString(attribute);
        } catch (Exception ex) {
            log.error("Writing error: {}", attribute);
        }
        return text;
    }

    @SuppressWarnings("unchecked")
    @Override
    public HashMap<String, Object> convertToEntityAttribute(String dbData) {
        HashMap<String, Object> map = new HashMap<>();
        try {
            map = objectMapper.readValue(dbData, map.getClass());
        } catch (Exception ex) {
            log.error("Reading error: {}", dbData);
        }
        return map;
    }
}

File: core/src/main/kotlin/com/tek/core/converter/HashMapConverter.kt
package com.tek.core.converter

import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import com.tek.core.util.LoggerDelegate
import javax.persistence.AttributeConverter

/**
 * Hashmap String Type Conversion.
 *
 *
 * This implementation converts a string representation of a map in a database (like plain JSON objects)
 * to a more friendly user HashMap java type.
 */
class HashMapConverter : AttributeConverter<HashMap<String, Any>, String> {

    private val log by LoggerDelegate()

    override fun convertToDatabaseColumn(attribute: HashMap<String, Any>): String {
        var jsonData = ""
        try {
            jsonData = jacksonObjectMapper().writeValueAsString(attribute)
        } catch (ex: Exception) {
            log.error("JSON writing error: {}", attribute)
        }
        return jsonData
    }

    override fun convertToEntityAttribute(dbData: String?): HashMap<String, Any> {
        var mapData = hashMapOf<String, Any>()
        try {
            mapData = jacksonObjectMapper().readValue(dbData, mapData::class.java)
        } catch (ex: Exception) {
            log.error("JSON reading error: {}", dbData)
        }
        return mapData
    }
}